File: c++/bubble.cpp
//bubble sort by template
#include<iostream>
using namespace std;
template<typename n>
void bubblesort(n arr[],int size)
{
    int i,j;
    n temp;
    for(i=0;i<size;i++)
    {
        for(j=0;j<size-1;j++)
        {
            if(arr[j]<arr[j+1])
            {
                temp=arr[j];
                arr[j]=arr[j+1];
                arr[j+1]=temp;
            }
        }
    }
}
int main()
{
    int arr[10]={1,20,42,12,3,13,50,56,40,10};
    int size=10;
    cout<<"array before shorting"<<endl;
     for(int i=0;i<size;i++)
     {
        cout<<arr[i]<<"\t";
     }
     cout<<endl;
     bubblesort<int>(arr,size);
     cout<<"array after shorting\n";
     for(int i=0;i<size;i++)
     {
        cout<<arr[i]<<"\t";
     }
     cout<<endl;
}

File: java/bubble.java
import java.util.*;
public class bubble {
    public static void main(String args[]){
        int size;
        Scanner sc= new Scanner(System.in);

        System.out.println("enetr size");
        size=sc.nextInt();

        int arr[]=new int[size];
        int i,j,temp;
        System.out.println("enter"+size+"value");
        for(i=0;i<size;i++){
            arr[i]=sc.nextInt();

        }
        for(i=1;i<size-1;i++){
            for(j=i-1;j<size;j++){
                if(arr[i]>arr[j]){
                    temp=arr[i];
                    arr[i]=arr[j];
                    arr[j]=temp;
                }

            }

        }
        System.out.println(" array after sorting");
        for(i=0;i<size;i++){
        System.out.println(arr[i]+" ");
        }  
    }
    
}
